core/src/main/java/org/commonjava/maven/galley/io/SpecialPathManagerImpl.java

```diff
@@ -50,17 +53,25 @@
 
     private Map<String, SpecialPathSet> pkgtypes;
 
+    private CacheProvider cacheProvider;
```

Comment:
Okay, I see the dependency cycle between components when you pair this with the various TransferDecorator implementations, and the CacheProvider. I wonder if it would make sense for this class to use the PathGenerator directly, in that case? That would break the dependency loop, I think.

Reply:
Agree :+1: Changed.